lib/core/include/key_value_proxy.hpp

```diff
@@ -0,0 +1,540 @@
+            /// \returns iterator
+            ///
+            /// \since 4.2.8
+            auto operator++() -> iterator
+            {
+                if (++index_ >= kvp_->len) {
+                    index_ = -1;
+                }
+                return *this;
+            }
+
+            /// \fn iterator operator++(int)
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \returns iterator
+            ///
+            /// \since 4.2.8
+            auto operator++(int) -> iterator
+            {
+                iterator ret = *this;
+                ++(*this);
+                return ret;
+            }
+
+            /// \fn bool operator==(const iterator& _rhs) const
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \returns bool
+            /// \retval true if indices match; otherwise, false.
+            ///
+            /// \since 4.2.8
+            auto operator==(const iterator& _rhs) const -> bool
+            {
+                return index_ == _rhs.index_;
+            }
+
+            /// \fn bool operator!=(const iterator& _rhs) const
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \returns bool
+            /// \retval Inverse of operator==
+            ///
+            /// \since 4.2.8
+            auto operator!=(const iterator& _rhs) const -> bool
+            {
+                return !(*this == _rhs);
+            }
+
+            /// \fn handle operator*()
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \returns handle - See handle class for details
+            ///
+            /// \since 4.2.8
+            auto operator*() -> handle
+            {
+                return {kvp_->keyWord[index_], *kvp_};
+            }
+
+            /// \fn const handle operator*() const
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \returns handle - See handle class for details
+            ///
+            /// \since 4.2.8
+            auto operator*() const -> const handle
+            {
+                return {kvp_->keyWord[index_], *kvp_};
+            }
+
+        private:
+            /// \var size_type index_
+            ///
+            /// \brief Index into the array of kvp
+            ///
+            /// \since 4.2.8
+            size_type index_;
+
+            /// \var kvp* kvp_
+            ///
+            /// \brief Pointer to proxy object's underlying kvp
+            ///
+            /// \since 4.2.8
+            kvp* kvp_;
+        };
+
+        /// \fn key_value_proxy(kvp& _kvp)
+        ///
+        /// \brief Constructs proxy using an existing kvp
+        ///
+        /// \since 4.2.8
+        key_value_proxy(kvp& _kvp)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'key_value_proxy' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/trel/irods/pullRequest?prid=4508572)